client/src/containers/widget/index.tsx

```diff
@@ -1,29 +1,37 @@
 "use client";
 import { useState } from "react";
 
-import { WidgetVisualizationsType } from "@shared/dto/widgets/widget-visualizations.constants";
+import {
+  WidgetData,
+  WidgetNavigationData,
+} from "@shared/dto/widgets/base-widget-data.interface";
+import {
+  WIDGET_VISUALIZATIONS,
+  WidgetVisualizationsType,
+} from "@shared/dto/widgets/widget-visualizations.constants";
 
-import { cn } from "@/lib/utils";
+import { cn, isWidgetData, isWidgetNavigationData } from "@/lib/utils";
 
 import AreaChart from "@/containers/widget/area-chart";
 import HorizontalBarChart from "@/containers/widget/horizontal-bar-chart";
+import Navigation from "@/containers/widget/navigation";
 import PieChart from "@/containers/widget/pie-chart";
 import SingleValue from "@/containers/widget/single-value";
 import WidgetHeader from "@/containers/widget/widget-header";
 
 import { Card } from "@/components/ui/card";
-import { WidgetData } from "@/types";
 
-interface WidgetProps {
+export interface WidgetProps {
   indicator: string;
   question: string;
   defaultVisualization: WidgetVisualizationsType;
   visualisations: WidgetVisualizationsType[];
-  data: WidgetData;
+  data: WidgetData | WidgetNavigationData;
```

Comment:
Can you use `BaseWidgetWithData['data]` instead? I think it reflects better the nature of data in this case.